dhis-2/dhis-services/dhis-service-setting/src/main/java/org/hisp/dhis/setting/SystemSettingManager.java

```diff
@@ -73,11 +78,10 @@
      * @param key the system setting key.
      * @return the setting value.
      */
-    Serializable getSystemSetting( SettingKey key );
-
+    @Transactional( readOnly = true )
     default <T extends Serializable> T getSystemSetting( SettingKey key, Class<T> as )
     {
-        return as.cast( getSystemSetting( key ) );
+        return as.cast( getSystemSetting( key, key.getDefaultValue() ) );
     }
```

Comment:
Could we call the class type parameter "type" instead of "as"? Would be nice to standardize on this, we already have "clazz", "klass", etc.

Reply:
done